analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Tests/VerifierTest.cs

```diff
@@ -159,6 +159,47 @@ public class Sample
         public void Verify_Cshtml() =>
             DummyWithLocationMapping.AddPaths("Dummy.cshtml").Verify();
 
+        [TestMethod]
+        public void Verify_Razor_WithFramework()
+        {
+            var verifierBuilder = DummyWithLocationMapping.AddPaths("Dummy.razor");
+            verifierBuilder.WithFramework("net6.0")
+                           .Invoking(x => x.Verify())
+                           .Should()
+                           .NotThrow();
+        }
+
+        [TestMethod]
+        public void Verify_Razor_WithFramework_NotSupported()
+        {
+            var verifierBuilder = DummyWithLocationMapping.AddPaths("Dummy.razor");
+            var frameworks = new string[]
+            {
+                "net48",
+                "netcoreapp3.1",
+                "netstandard2.1",
+                "net5.0"
+            };
+
+            foreach (var framework in frameworks)
+            {
+                verifierBuilder.WithFramework(framework)
+                           .Invoking(x => x.Verify())
+                           .Should()
+                           .Throw<InvalidOperationException>().WithMessage("Razor compilation is supported starting from .NET 6 and onwards.");
+            }
+        }
```

Comment:
You can use `[DataRow]` to have more granular test cases:
```suggestion
        [DataTestMethod]
        [DataRow("net48")]
        [DataRow("netcoreapp3.1")]
        [DataRow("netstandard2.1")]
        [DataRow("net5.0")]
        public void Verify_Razor_WithFramework_NotSupported(string framework)
        {
            var verifierBuilder = DummyWithLocationMapping.AddPaths("Dummy.razor");

            verifierBuilder.WithFramework(framework)
                .Invoking(x => x.Verify())
                .Should()
                .Throw<InvalidOperationException>().WithMessage("Razor compilation is supported starting from .NET 6 and onwards.");
        }
```